common/flatpak-dir.c

```diff
@@ -11073,6 +11073,45 @@ flatpak_dir_find_local_related (FlatpakDir *self,
   return g_steal_pointer (&related);
 }
 
+static gboolean
+flatpak_dir_get_deployed_metadata (FlatpakDir    *self,
+                                   const char    *ref,
+                                   GCancellable  *cancellable,
+                                   GKeyFile     **out_metadata,
+                                   GError       **error)
```

Comment:
I'm not sure why you need another function for this. Can't we use flatpak_deploy_get_metadata() & co.

Reply:
I think in this case I wanted to be able to share the logic between both the local and remote case, which would mean that they both used a function that used the `GKeyFile` structure. However, see below, it may not be necessary.